builtin/docker/builder.go

```diff
@@ -424,14 +436,28 @@ func (b *Builder) buildWithDocker(
 		return status.Errorf(codes.InvalidArgument, "buildkit is required to use platform option")
 	}
 
+	var authMap = make(map[string]types.AuthConfig)
+	if b.config.Auth != nil {
+		authMap[authConfig["host_name"]] = types.AuthConfig{
+			Username:      authConfig["user_name"],
+			Password:      authConfig["password"],
+			Email:         authConfig["email"],
+			Auth:          authConfig["auth"],
+			ServerAddress: authConfig["server_address"],
+			IdentityToken: authConfig["identity_token"],
+			RegistryToken: authConfig["registry_token"],
```

Comment:
`authConfig` should be a struct instead of a map to avoid folks setting fields that are ignored.